### PR TITLE
Add Integrations page for Go

### LINKED ISSUE #6229: Add integrations docs
We currently do not have a dedicated page on our Go docs that tell people which integrations are enabled in the SDK nor how to disable some of them.

Should look similar to something like https://docs.sentry.io/platforms/php/integrations/

Refs https://github.com/getsentry/sentry-go/issues/571

### PR DESCRIPTION
**Pull request by ribice**
Resolves #6229

### CODE REVIEW

**Comment by ribice** [comment]
Don't close this

**Comment by cleptric** [comment]
@ribice you can add the WIP label, with it, it will stay open. Merging this next week.

File: docs/platforms/go/common/integrations/index.mdx [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+```go
+sentry.Init(sentry.ClientOptions{
+    Dsn:               "___PUBLIC_DSN___",
+	Integrations: func(i []Integration) []Integration {
+		return []Integration{}
+	    },
+    }
+)
+```
```
Comment (cleptric) [inline docs/platforms/go/common/integrations/index.mdx:21]: Indentation is off.

File: docs/platforms/go/common/integrations/index.mdx [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+### ModulesIntegration
+
+This integration logs all the Go modules used in your project, including versions, as part of event details.
```
Comment (cleptric) [inline docs/platforms/go/common/integrations/index.mdx:25]:
```suggestion
This integration records all Go modules used in your application, including the version, as part of event details.
```

File: docs/platforms/go/common/integrations/index.mdx [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+### EnvironmentIntegration
+
+This integration fills the event data with Go runtime and OS-level information, such as CPU count, Go architecture, and runtime version.
```
Comment (cleptric) [inline docs/platforms/go/common/integrations/index.mdx:29]:
```suggestion
This integration fills the event data with Go runtime and OS-level specific information, such as CPU core/thread count, Go architecture, and runtime version.
```

Comment (cleptric) [reply docs/platforms/go/common/integrations/index.mdx:29]: Go architecture?

File: docs/platforms/go/common/integrations/index.mdx [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+
+This integration lets you ignore specific transactions that match the provided patterns. This is useful when certain transaction names should not be captured.
+
+### ContextifyFramesIntegration
+
+This integration captures context around lines of code that caused an error. It provides a snapshot of source code, including lines before and after the error, for better debugging.
+
+### GlobalTagsIntegration
+
+This integration adds global tags to all events. These can be defined as environment variables using the SENTRY_TAGS_ prefix or within the client options.
```
Comment (cleptric) [inline docs/platforms/go/common/integrations/index.mdx:45]:
```suggestion
This integration adds global tags to all events. These can be defined as environment variables using the `SENTRY_TAGS_` prefix or within the client options.
```

File: docs/platforms/go/common/integrations/index.mdx [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+		for _, integration := range integrations {
+			if integration.Name() == "ContextifyFrames" {
+				continue
+			}
+			filteredIntegrations = append(filteredIntegrations, integration)
+		}
+		return filteredIntegrations
+	},
+})
+```
```
Comment (cleptric) [inline docs/platforms/go/common/integrations/index.mdx:64]: Indentation is off.

Comment (ribice) [reply docs/platforms/go/common/integrations/index.mdx:63]: Sorry, not sure how this is off? Running it locally it looks like this:

image

File: docs/platforms/go/common/integrations/index.mdx (hunk 1 above)
Comment (ribice) [reply docs/platforms/go/common/integrations/index.mdx]: Same here:

image

**Review by cleptric — APPROVED** [review]

**Pull request merged**